### ISSUE TITLE
Deferred first tick may cause priority inversion.

### ISSUE THREAD

**Issue by mbostock** [d3/d3-transition#16]
Related mbostock/d3#1576.

Consider the looped transition from [bl.ocks.org/7df5eb012ff76589419d](http://bl.ocks.org/mbostock/7df5eb012ff76589419d):

```js
selection.transition()
    .duration(2500)
    .delay(function(d) { return d * 40; })
    .each("start", slideRight);

function slideRight() {
  d3.activeTransition(this)
      .attr("cx", width)
    .transition()
      .each("start", slideLeft);
}

function slideLeft() {
  d3.activeTransition(this)
      .attr("cx", 0)
    .transition()
      .each("start", slideRight);
}
```

Say the page is background during the slide-right transition. If the page is returned to the foreground after five seconds (greater than twice the 2500ms duration), then the slide-right transition immediately ends, setting all the circles to the far right. Then the slide-left transition starts. Simultaneously, a new slide-right transition is scheduled to follow the slide-left transition.

The problem is that when the slide-left transition starts, its first (and only) tick—when it would set the circles to the left—is *deferred* to the end of the frame (due to the optimization in mbostock/d3#1576). Thus, the following slide-right transition starts and initializes its tweens *before* the slide-left transition gets its tick, and the slide-right transition may do nothing for 2500ms:

![screen shot 2015-11-13 at 10 06 46 am](https://cloud.githubusercontent.com/assets/230541/11153519/5b2c7512-89ee-11e5-91ea-0177b11147b6.png)
 
Note that if we define the transition ever so differently, the stuttering does not occur:

```js
select.transition()
    .duration(2500)
    .delay(function(d) { return d * 40; })
    .attr("cx", width)
    .each("end", slideLeft);

function slideLeft() {
  d3.activeTransition(this).transition()
      .attr("cx", 0)
      .each("end", slideRight);
}

function slideRight() {
  d3.activeTransition(this).transition()
      .attr("cx", width)
      .each("end", slideLeft);
}
```

In this case, when slide-right is paused and resumed, the slide-right transition immediately ends, setting all the circles to the far right. Then the slide-left transition starts. The slide-left transition immediately ends on the following frame, setting the circles to the far left. Then a new slide-right transition starts. The relative ordering is correctly preserved.

**Comment by mbostock** [comment]
Fixed in 3.5.9.

**Comment by mbostock** [comment]
Somewhat amazingly, we can fix this simply by scheduling the deferred first tick invocation before dispatching the start event. This causes the deferred first tick to be invoked before any new transition that is scheduled during the start event, maintaining priority order.
